### ISSUE TITLE
Consider showing a type hint for function clauses

### ISSUE THREAD

**Issue by mmagic-boy** [JetBrains/intellij-arend#264]
Consider the following function:

```agda
\func -'+assoc (m n p : Nat): m -' n -' p = m -' (n + p)
  | m, n, 0 => [n-'0=n] (m -' n)
  | m, n, p => {?}
```

It is defined by pattern matching, so each clause has its own type. For example, the type of the first clause is `m -' n -' 0 = m -' n`. You cannot see that type when you read the code, and it could be a non trivial task to calculate it in your head. But knowing the type helps to understand why the proof on the right side of `=>` looks the way it does. It would be nice if IDE could show me the type explicitly.

One way to do that is to use inlay type hints. An example in Kotlin:

<img width="1024" alt="Screen Shot 2021-04-20 at 10 27 25 PM" src="https://user-images.githubusercontent.com/5653229/115452837-9c9ea900-a227-11eb-96da-bd4c2636dce9.png">

Here `: List<Path!>` is a hint that shows a type of the `foo` variable that is not specified explicitly. In Arend that might look like this:

```agda
\func -'+assoc (m n p : Nat): m -' n -' p = m -' (n + p)
  | m, n, 0 => [n-'0=n] (m -' n) : m -' n -' 0 = m -' n
  | m, n, p => {?}
```

**Comment by ice1000** [comment]
Currently you can select an Arend expression and do Ctrl Shift P to show its type

**Comment by mmagic-boy** [comment]
Yeah, I use it. I believe type hints might be more convenient when you mostly read code. Selecting and invoking the action for each clause is too tedious.

**Comment by ice1000** [comment]
FWIW the 'show type' action is kinda expensive

**Comment by mmagic-boy** [comment]
Oh, I see. That could be a problem, indeed.
